doc/source/ray-air/preprocessors.rst

```diff
@@ -0,0 +1,175 @@
+.. _air-preprocessors:
+
+Preprocessing Data
+==================
+
+This page describes how to perform data preprocessing in Ray AIR.
+
+Data preprocessing is a common technique for transforming raw data into features that will be input to a machine learning model.
+In general, you may want to apply the same preprocessing logic to your offline training data and online inference data.
+Ray AIR provides several common preprocessors out of the box as well as interfaces that enable you to define your own custom logic.
+
+Overview
+--------
+
+Ray AIR exposes a ``Preprocessor`` class for preprocessing. The Preprocessor has 4 methods that make up its core interface.
+
+#. ``fit()``: Compute some state for a Dataset and save it to the Preprocessor.
+#. ``transform()``: Apply a transformation to a Dataset.
+#. ``transform_batch()``: Apply a transformation to a single batch of data.
+#. ``fit_transform()``: Syntactic sugar for calling both ``fit()`` and ``transform()`` on a Dataset.
+
+To show these in action, let's walk through a basic example. First we'll set up 2 simple Ray Datasets.
+
+.. literalinclude:: doc_code/preprocessors.py
+    :language: python
+    :start-after: __preprocessor_setup_start__
+    :end-before: __preprocessor_setup_end__
+
+Next, ``fit`` the Preprocessor on one Dataset, and ``transform`` both Datasets with this fitted information.
+
+.. literalinclude:: doc_code/preprocessors.py
+    :language: python
+    :start-after: __preprocessor_fit_transform_start__
+    :end-before: __preprocessor_fit_transform_end__
+
+Finally, call ``transform_batch`` on a single batch of data.
```

Comment:
It's not entirely clear from this example what a "batch" is. So far, all data consists of 4 items. People might misinterpret batch for pandas or so. Why can't a Dataset be a batch and how do we differentiate this in this example?

Reply:
Updated the first reference to batch to point to `DataBatchType`, which should hopefully make it more clear that this is for Predictors.